### PR TITLE
WCMS-20047: Data Dictionary Documentation

### PR DESCRIPTION
**Pull request by rovcase**
Fixes [issue#]

## QA Steps

- [ ] Add manual QA steps in checklist format for a reviewer to perform. Be as specific as possible, provide examples if appropriate.

## Checklist before requesting review

_If any of these are left unchecked, please provide an explanation_

- [ ] I have updated or added documentation

### CODE REVIEW

File: docs/source/user-guide/guide_data_dictionaries.rst [hunk 1]
```diff
@@ -195,13 +195,38 @@ Creating a data dictionary via the UI
...
-7. Click the "Save" button.
-8. See a list of your data dictionaries at `/api/1/metastore/schemas/data-dictionary/items/`
-9. Edit your data dictionary by going to `/admin/dkan/data-dictionaries`.
-10. Click the "Edit" link in the right-hand column next to the data dictionary you want to edit.
+4. To define a "Data Dictionary field" start by clicking "Add field".
+
+  .. image:: images/fields/add-data-field.png
+    :alt: A portion of the data dictionary creation admin page with a focus on the "Add field" button.
+
+5. An "Add new field" editable form box will appear. Complete the required form fields here, reference the Table Schema section above if needed.
```
Comment (dmundra) [inline docs/source/user-guide/guide_data_dictionaries.rst]: Nitpick, quote the 'Table Schema' and link it to that header in the page?

File: docs/source/user-guide/guide_data_dictionaries.rst [hunk 2]
```diff
@@ -195,13 +195,38 @@ Creating a data dictionary via the UI
...
+
+  .. image:: images/fields/add-data-field-editable-form-box.png
+    :alt: A portion of the data dictionary creation admin page with a focus on the "Add new field" section.
```
Comment (dmundra) [inline docs/source/user-guide/guide_data_dictionaries.rst:206]: Love the alt text. Thank you!

File: docs/source/user-guide/guide_data_dictionaries.rst [hunk 3]
```diff
@@ -195,13 +195,38 @@ Creating a data dictionary via the UI
...
+7. Alternatively you can click "Cancel" to stop creating the Data Dictionary field, at which point all the information you entered will be removed and the "Add new Field" box will close.
+8. Note that after saving, an informational table will appear listing the data dictionary fields you have created thus far and their names, titles, and details (data types, formats and descriptions).
+
+  .. image:: images/fields/edit-data-field-informational-table.png
+    :alt: A portion of the data dictionary creation admin page with a focus on the informational table that appears.
+
+9. You also have the ability to edit or modify the data dictionary fields you've created so far by clicking the gear icon on the right-hand side of this table.
+10.  Clicking the gear icon will open an editable form that will show the previously entered information for the respective data dictionary field. You can then modify this information and either: save (retain the modifications), cancel (close the editable form box without retaining any modifications) or delete (remove the respective Data Dictionary field entirely).
```
Comment (dmundra) [inline docs/source/user-guide/guide_data_dictionaries.rst]: I would maybe quote the 'save', 'cancel', and 'delete' or at least title case it. Nit pick though.

Comment (dmundra) [reply docs/source/user-guide/guide_data_dictionaries.rst]: This will match the indexes section where you use double quotes and title case.

**Review by dmundra — CHANGES_REQUESTED** [review]
@rovcase looking good. Left you comments to consider for the data dictionary page. Let me know what you think.

File: docs/source/user-guide/guide_data_dictionaries.rst (hunk 1 above)
Comment (rovcase) [reply docs/source/user-guide/guide_data_dictionaries.rst]: Good call, fixed. Thanks!

File: docs/source/user-guide/guide_data_dictionaries.rst (hunk 3 above)
Comment (rovcase) [reply docs/source/user-guide/guide_data_dictionaries.rst]: Agreed, ty. Updated!

**Comment by rovcase** [comment]
> @rovcase looking good. Left you comments to consider for the data dictionary page. Let me know what you think.

Tysm for the look over, added your changes and should be good now.

File: docs/source/user-guide/guide_data_dictionaries.rst (hunk 3 above)
Comment (dmundra) [reply docs/source/user-guide/guide_data_dictionaries.rst]: @rovcase Cancel and delete look good but there seems to be an [extra line break now](https://github.com/GetDKAN/dkan/pull/4271/files#diff-f8f5d58adf4db85409826bcba707e6ebe30dde7abe1030fc2459440daa71820fR216-R217) after "Cancel" and also need to apply the same change to "Save".

**Comment by dmundra** [comment]
> > @rovcase looking good. Left you comments to consider for the data dictionary page. Let me know what you think.
> 
> Tysm for the look over, added your changes and should be good now.

Thank you @rovcase. I think one more fix (I left a reply on a previous comment) and you should be good to go.

**Pull request merged**